### PR TITLE
Added a catch for non secure context on clipboard copy

### PR DESCRIPTION
**Pull request by sourcesoldier**

### CODE REVIEW

File: src/Resources/js/modules/custom-functions.js [hunk 1]
```diff
@@ -40,7 +40,10 @@ Mealz.prototype.copyToClipboard = function() {
             } else {
                 guestMenuLinkInput.trigger('select');
             }
-            navigator.clipboard.writeText(result);
+            // Clipboard copy only works in secure context, redirect http -> https
+            if(window.isSecureContext) {
+                navigator.clipboard.writeText(result);
+            }
```
Comment (chetan-thapliyal) [inline src/Resources/js/modules/custom-functions.js:46]: We can improve it a bit more by adding "console.warn('Clipboard access from insecure (HTTP) context is prohibited. Reload page with HTTPS and try again.')" in else block. The comment above can then be removed.

Comment (tgietzen) [reply src/Resources/js/modules/custom-functions.js:46]: +1

**Review by tgietzen — APPROVED** [review]

File: src/Resources/js/modules/custom-functions.js [hunk 2]
```diff
@@ -40,7 +40,12 @@ Mealz.prototype.copyToClipboard = function() {
             } else {
                 guestMenuLinkInput.trigger('select');
             }
-            navigator.clipboard.writeText(result);
+            // Clipboard copy only works in secure context, redirect http -> https
+            if(window.isSecureContext) {
```
Comment (hacksch) [inline src/Resources/js/modules/custom-functions.js:44]: this is not supported in normal internet explorers
https://developer.mozilla.org/en-US/docs/Web/API/isSecureContext#browser_compatibility

alternative you can check with
`
window.location.protocol == 'https:'
`

Comment (tgietzen) [reply src/Resources/js/modules/custom-functions.js:44]: Done.

**Pull request merged**